### DISCUSSION TITLE
Handlers or SetupServer()

### DISCUSSION THREAD

**Discussion by charlestbell** [mswjs/msw#1003] [Q&A] [answered]
Which one should I use? Handlers or SetupWorker()? It is not clear in the docs.

The docs mention both [Handlers](https://mswjs.io/docs/getting-started/mocks/rest-api)
and [SetupServer()](https://mswjs.io/docs/api/setup-server) with no explanation or comparison between them. Which do I use in which situation?

I'm building a react-native app.

**Comment by kettanaito** [comment, marked as answer]
Hi, @charlestbell. Sorry about the confusion. 

Handlers (or "request handlers") are functions that describe what requests to capture and how to mock their responses. Think of it as a network behavior description. Handlers do _not_ do any interception/mocking on their own, they are but a set of instructions on what to do when a matching request happens. 

APIs like `setupServer` and `setupWorker` **enable interception of requests**. Calling them is what makes MSW aware of the requests that happen in your application. As you've seen from the [Getting started tutorial](https://mswjs.io/docs/getting-started/mocks), you provide request handlers to one of those APIs (or both!) depending on the environment you wish to enable mocks in. 

As you're building a React Native application, you'll be using `setupServer` (React Native is still Node.js).
